node/src/components/contract_runtime/operations.rs

```diff
@@ -29,14 +29,19 @@ use crate::{
     types::{Block, Deploy, DeployHash, DeployHeader, FinalizedBlock},
 };
 
-pub(super) fn execute_finalized_block(
+pub(super) async fn execute_finalized_block(
```

Comment:
This method  doesn't have to be `async`. If you want to call it in the async context it should be lifted to an async one on the call-side rather than forcing all callers to handle it as async (even though it doesn't have to be).

Reply:
Thanks, an earlier version had required it be async, but you are correct.